pkg/store/tsdb.go

```diff
@@ -76,6 +77,9 @@ func NewTSDBStore(logger log.Logger, db TSDBReader, component component.StoreAPI
 }
 
 func (s *TSDBStore) SetExtLset(extLset labels.Labels) {
+	s.mtx.Lock()
+	defer s.mtx.Unlock()
```

Comment:
Should we use the mutex when reading labels? We can use the `RLock()` and `RUnlock` methods when reading.

Reply:
yeah that makes sense